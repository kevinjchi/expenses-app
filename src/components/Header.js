import React from 'react';
import { BrowserRouter, Route, Switch, Link , NavLink} from 'react-router-dom';

const Header = () => (

    // No full page refresh, do javascript rendering instead
    <header>
        <h1>expenses app </h1>
        <NavLink to="/" activeClassName="is-active" exact={true}> go home </NavLink>
        <NavLink to="/create" activeClassName="is-active"> create </NavLink>
        <NavLink to="/help" activeClassName="is-active"> help page </NavLink>
    </header>
);

export default Header;