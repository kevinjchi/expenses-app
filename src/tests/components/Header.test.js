import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';
import { shallow } from 'enzyme';

// snapshot testing of Header components

test('should render header correctly', () => {
    const wrapper = shallow( <Header /> );
    expect(wrapper.find('h1').length).toBe(1);
    // const renderer = new ReactShallowRenderer(); 
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
})

test('should render header correctly', () => {
    const wrapper = shallow( <Header /> );
    expect(wrapper).toMatchSnapshot();
    // const renderer = new ReactShallowRenderer(); 
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
})