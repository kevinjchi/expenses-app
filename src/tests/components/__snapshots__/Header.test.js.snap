// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    expenses app 
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
     go home 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
     create 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
     help page 
  </NavLink>
</header>
`;
